Issue 3185

mbukatov opened the pull request:
Based on agreement reached during last Weekly OCS Automation Sycnup (on 2020-10-22), this pull request is fixing issue https://github.com/red-hat-storage/ocs-ci/issues/3185 by tackling 1st part of issue https://github.com/red-hat-storage/ocs-ci/issues/408 as drafted in https://github.com/red-hat-storage/ocs-ci/issues/408#issuecomment-713827775:

- all non test code were moved into new helpers module `ocs_ci.helpers`
- in tests directory, there should be only tests or fixtures
- fixes few cycled deps
- adding new unittests, which catches such issues (non test code in tests dir, cycled deps) in the future

We can be sure that this change doesn't break something because:

- all unit tests passed, including new unit test which imports all ocs_ci modules
- flake8 check passed
- tox collectonly passed (so there are no import issues in test code neither)

clacroix12 commented on ocs_ci/tests/test_import_all.py:
```diff
@@ -0,0 +1,29 @@
+    For given module, return list of full module path names for all submodules
+    recursively.
+    """
+    return [name for _, name, _ in
+            pkgutil.walk_packages(path=[module_path], prefix=module_prefix)]
```
```suggestion
    return [
        name for _, name, _ in
        pkgutil.walk_packages(path=[module_path], prefix=module_prefix)
    ]
```

clacroix12 replied:
Looks like this change was reverted, this can be fixed after this PR is merged tho.

ebenahar reviewed (dismissed):
Much needed

clacroix12 reviewed (dismissed):
Looks great to me! @mbukatov just needs a rebase to resolve a merge conflict. Also there is one issue with the docs generation:

```
Warning, treated as error:

/home/travis/build/red-hat-storage/ocs-ci/ocs_ci/helpers/cluster_exp_helpers.py:docstring of ocs_ci.helpers.cluster_exp_helpers.ClusterFiller.filler:3:Unexpected indentation.

Makefile.tox:20: recipe for target 'html' failed

make: *** [html] Error 2

ERROR: InvocationError for command /usr/bin/make -f Makefile.tox html (exited with code 2)
```

@vasukulkarni I don't see any test failures that look like they were caused by the changes here, but please correct me if I'm wrong. If the test results look acceptable let's re-approve and merge once the conflict is resolved.

vasukulkarni commented:
There is actually a docs failure and I think one more rebase is needed due to conflict :) 

```
Warning, treated as error:

/home/travis/build/red-hat-storage/ocs-ci/ocs_ci/helpers/cluster_exp_helpers.py:docstring of ocs_ci.helpers.cluster_exp_helpers.ClusterFiller.filler:3:Unexpected indentation.

Makefile.tox:20: recipe for target 'html' failed
```

mbukatov commented:
Rebased to resolve conflicts, fixed syntax errors in docstrings of helpers module.

Executed a tox locally:

```
  flake8: commands succeeded
  docs: commands succeeded
  collectonly: commands succeeded
  py37: commands succeeded
```

mbukatov commented:
Marking as verified based on run from https://github.com/red-hat-storage/ocs-ci/pull/3211#pullrequestreview-518340948 (no evidence of import errors or helpers failures noticed).
